modules/pytket-stim/pytket/extensions/stim/backends/stim_backend.py

```diff
@@ -0,0 +1,182 @@
+# Copyright 2020-2021 Cambridge Quantum Computing
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import cast, List, Optional, Sequence, Union
+from uuid import uuid4
+from pytket.backends import (
+    Backend,
+    CircuitNotRunError,
+    CircuitStatus,
+    ResultHandle,
+    StatusEnum,
+)
+from pytket.backends.backendresult import BackendResult
+from pytket.backends.resulthandle import _ResultIdTuple
+from pytket.circuit import Circuit, OpType  # type: ignore
+from pytket.passes import (  # type: ignore
+    BasePass,
+    DecomposeBoxes,
+    FlattenRegisters,
+    RebaseCustom,
+    RemoveRedundancies,
+    SequencePass,
+)
+from pytket.predicates import (  # type: ignore
+    DefaultRegisterPredicate,
+    GateSetPredicate,
+    NoClassicalControlPredicate,
+    Predicate,
+)
+from pytket.utils.outcomearray import OutcomeArray
+from pytket.utils.results import KwargTypes
+import numpy as np
+import stim  # type: ignore
+
+_gate = {
+    OpType.noop: "I",
+    OpType.X: "X",
+    OpType.Y: "Y",
+    OpType.Z: "Z",
+    OpType.H: "H",
+    OpType.S: "S",
+    OpType.SX: "SQRT_X",
+    OpType.SXdg: "SQRT_X_DAG",
+    OpType.CX: "CX",
+    OpType.CY: "CY",
+    OpType.CZ: "CZ",
+    OpType.ISWAPMax: "ISWAP",
+    OpType.SWAP: "SWAP",
+    OpType.Measure: "M",
+    OpType.Reset: "R",
+}
+
+
+def _int_double(x: float) -> int:
+    # return (2x) mod 8 if x is close to a half-integer, otherwise error
+    y = 2 * x
+    n = int(np.round(y))  # type: ignore
+    if np.isclose(y, n):
```

Comment:
What is the tolerance on isclose?

Reply:
By default, `rtol=1e-05, atol=1e-08` and 
"The relative difference (`rtol` * abs(`b`)) and the absolute difference `atol` are added together to compare against the absolute difference between `a` and `b`."
The second value (here `n`) is the reference value.

Reply:
ty